### ISSUE TITLE
System.NullReferenceException Error When Using apireference

### ISSUE THREAD

**Issue by RealDotNetDave** [ap0llo/mddocs#73]
I just tried your tool and am getting the error below. Is there something I'm doing wrong?

```
C:\src\GitHub\dotNetTips.Utility.Core\src>mddocs apireference --assembly C:\src\GitHub\dotNetTips.Utility.Core\appbin\netcoreapp2.2\dotNetTips.Utility.Standard.Extensions.dll --outdir c:\temp
INFORMATION: Using preset 'Default' for generating markdown
INFORMATION: Loading assembly from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netcoreapp2.2\dotNetTips.Utility.Standard.Extensions.dll'
INFORMATION: Reading XML documentation comments from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netcoreapp2.2\dotNetTips.Utility.Standard.Extensions.xml'
WARNING: Encountered unknown element 'count'. Element will be ignored.
WARNING: Encountered unknown element 'count'. Element will be ignored.
WARNING: Encountered unknown element 'count'. Element will be ignored.
WARNING: Encountered unknown element 'count'. Element will be ignored.
WARNING: Encountered unknown element 'count'. Element will be ignored.
Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at Grynwald.MdDocs.ApiReference.Model.CSharpDefinitionFormatter.GetLiteral(TypeReference typeReference, Object value) in D:\a\1\s\src\MdDocs.ApiReference.Model\_Helpers\CSharpDefinitionFormatter.cs:line 693
   at Grynwald.MdDocs.ApiReference.Model.CSharpDefinitionFormatter.GetDefinition(ParameterDefinition parameter) in D:\a\1\s\src\MdDocs.ApiReference.Model\_Helpers\CSharpDefinitionFormatter.cs:line 759
   at System.Linq.Enumerable.SelectIListIterator`2.MoveNext()
   at System.Text.StringBuilder.AppendJoinCore[T](Char* separator, Int32 separatorLength, IEnumerable`1 values)
   at System.Text.StringBuilder.AppendJoin[T](String separator, IEnumerable`1 values)
   at Grynwald.MdDocs.ApiReference.Model.CSharpDefinitionFormatter.GetDefinition(MethodDefinition method) in D:\a\1\s\src\MdDocs.ApiReference.Model\_Helpers\CSharpDefinitionFormatter.cs:line 234
   at Grynwald.MdDocs.ApiReference.Model.MethodLikeOverloadDocumentation..ctor(MethodDefinition definition, IXmlDocsProvider xmlDocsProvider) in D:\a\1\s\src\MdDocs.ApiReference.Model\MethodLikeOverloadDocumentation.cs:line 63
   at Grynwald.MdDocs.ApiReference.Model.MethodOverloadDocumentation..ctor(MethodDocumentation methodDocumentation, MethodDefinition definition, IXmlDocsProvider xmlDocsProvider) in D:\a\1\s\src\MdDocs.ApiReference.Model\MethodOverloadDocumentation.cs:line 25
   at Grynwald.MdDocs.ApiReference.Model.MethodDocumentation.<>c__DisplayClass7_0.<.ctor>b__1(MethodDefinition d) in D:\a\1\s\src\MdDocs.ApiReference.Model\MethodDocumentation.cs:line 42
   at System.Linq.Enumerable.SelectIListIterator`2.MoveNext()
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector)
   at Grynwald.MdDocs.ApiReference.Model.MethodDocumentation..ctor(TypeDocumentation typeDocumentation, IEnumerable`1 definitions, IXmlDocsProvider xmlDocsProvider) in D:\a\1\s\src\MdDocs.ApiReference.Model\MethodDocumentation.cs:line 41
   at Grynwald.MdDocs.ApiReference.Model.TypeDocumentation.<>c__DisplayClass91_0.<.ctor>b__15(IGrouping`2 group) in D:\a\1\s\src\MdDocs.ApiReference.Model\TypeDocumentation.cs:line 242
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey](IEnumerable`1 source, Func`2 keySelector)
   at Grynwald.MdDocs.ApiReference.Model.TypeDocumentation..ctor(ModuleDocumentation moduleDocumentation, NamespaceDocumentation namespaceDocumentation, TypeDefinition definition, IXmlDocsProvider xmlDocsProvider, ILogger logger, TypeDocumentation declaringType) in D:\a\1\s\src\MdDocs.ApiReference.Model\TypeDocumentation.cs:line 239
   at Grynwald.MdDocs.ApiReference.Model.ModuleDocumentation.LoadTypeRecursively(TypeDefinition typeDefinition, TypeDocumentation declaringType) in D:\a\1\s\src\MdDocs.ApiReference.Model\ModuleDocumentation.cs:line 119
   at Grynwald.MdDocs.ApiReference.Model.ModuleDocumentation..ctor(AssemblyDocumentation assemblyDocumentation, ModuleDefinition definition, IXmlDocsProvider xmlDocsProvider, ILogger logger) in D:\a\1\s\src\MdDocs.ApiReference.Model\ModuleDocumentation.cs:line 61
   at Grynwald.MdDocs.ApiReference.Model.AssemblyDocumentation..ctor(AssemblyDefinition definition, IXmlDocsProvider xmlDocsProvider, ILogger logger) in D:\a\1\s\src\MdDocs.ApiReference.Model\AssemblyDocumentation.cs:line 42
   at Grynwald.MdDocs.ApiReference.Model.AssemblyDocumentation.FromAssemblyFile(String filePath, ILogger logger) in D:\a\1\s\src\MdDocs.ApiReference.Model\AssemblyDocumentation.cs:line 83
   at Grynwald.MdDocs.Program.OnApiReferenceCommand(ILogger logger, ApiReferenceOptions opts) in D:\a\1\s\src\MdDocs\Program.cs:line 67
   at Grynwald.MdDocs.Program.<>c.<Main>b__0_1(ApiReferenceOptions opts) in D:\a\1\s\src\MdDocs\Program.cs:line 31
   at CommandLine.ParserResultExtensions.MapResult[T1,T2,TResult](ParserResult`1 result, Func`2 parsedFunc1, Func`2 parsedFunc2, Func`2 notParsedFunc)
   at Grynwald.MdDocs.Program.Main(String[] args) in D:\a\1\s\src\MdDocs\Program.cs:line 28
```

**Comment by ap0llo** [comment]
Which version are you using? I think I ran into that issue myself and fixed it.
Could you please try the latest CI build from the [MyGet feed](https://www.myget.org/feed/ap0llo-mddocs/package/nuget/Grynwald.MdDocs)?

**Comment by RealDotNetDave** [comment]
I installed your global tool as in your documentation. Not sure what version it is. I tried installing the package you recommended and got this error:

```
Install-Package : Unable to find package source 'https://www.myget.org/F/ap0llo-mddocs/api/v3/index.json'. Use Get-PackageSource 
to see all available package sources.
At line:1 char:1
+ Install-Package Grynwald.MdDocs -Version 0.4.156-pre -Source https:// ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power....InstallPackage:InstallPackage) [Install-Package], Exception
    + FullyQualifiedErrorId : SourceNotFound,Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage
```

**Comment by ap0llo** [comment]
You can check the version you're running using `mddocs --version`.

To install the latest prerelease version, try:

```
dotnet tool install -g Grynwald.MdDocs --version 0.4.156-pre --add-source https://www.myget.org/F/ap0llo-mddocs/api/v3/index.json 
```

**Comment by RealDotNetDave** [comment]
When I do that it says it's already installed. I tried this but didn't work. Any ideas?

`dotnet tool uninstall Grynwald.MdDocs`

David

**Comment by RealDotNetDave** [comment]
I got it working with update. But I see the info below. Is this okay? Also, is there any way to create one MD doc for an entire assembly?

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview>mddocs apireference --assembly C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.dll --outdir c:\temp\mdoc
INFORMATION - Loading assembly from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.dll'
INFORMATION - Reading XML documentation comments from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.xml'
WARNING - Encountered unknown element 'count'. Element will be ignored.
WARNING - Encountered unknown element 'count'. Element will be ignored.
WARNING - Encountered unknown element 'count'. Element will be ignored.
WARNING - Encountered unknown element 'count'. Element will be ignored.
WARNING - Encountered unknown element 'count'. Element will be ignored.
INFORMATION - Using preset 'Default' for generating markdown
```

**Comment by ap0llo** [comment]
> I got it working with update. But I see the info below. Is this okay? Also, is there any way to create one MD doc for an entire assembly?

No there isn't such an option at the moment. MdDocs will always produce multiple output files.

> 
> ```
> C:\Program Files (x86)\Microsoft Visual Studio\2019\Preview>mddocs apireference --assembly C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.dll --outdir c:\temp\mdoc
> INFORMATION - Loading assembly from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.dll'
> INFORMATION - Reading XML documentation comments from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Extensions.xml'
> WARNING - Encountered unknown element 'count'. Element will be ignored.
> WARNING - Encountered unknown element 'count'. Element will be ignored.
> WARNING - Encountered unknown element 'count'. Element will be ignored.
> WARNING - Encountered unknown element 'count'. Element will be ignored.
> WARNING - Encountered unknown element 'count'. Element will be ignored.
> INFORMATION - Using preset 'Default' for generating markdown
> ```

The warning you see in the log indicates that the XML documentation file of the assembly contains some `count` elements which are not recognized. Aside from the contents of these tags, the documentation should have been generated just fine.

**Comment by RealDotNetDave** [comment]
Hi. I had to refresh Windows which meant I had to re-install Visual Studio and I'm having an issue running MdDocs again. When I try to uninstall it I get an error saying it can't find the manifest file. Just for your info, here is the error I am seeing when I do run it.

```
C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0>mddocs apireference --assembly C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Common.dll --outdir c:\temp\mdoc
INFORMATION - Loading assembly from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Common.dll'
INFORMATION - Reading XML documentation comments from 'C:\src\GitHub\dotNetTips.Utility.Core\appbin\netstandard2.0\dotNetTips.Utility.Standard.Common.xml'
Unhandled exception. System.ArgumentException: A document with path 'dotNetTips/Utility/Standard/Common/ControlChars/fields/CR.md' already exists. (Parameter 'path')
   at Grynwald.MarkdownGenerator.DocumentSet`1.Add(String path, T document)
   at Grynwald.MdDocs.ApiReference.Templates.Default.PageFactory.LoadPages() in /_/src/MdDocs.ApiReference/Templates/Default/PageFactory.cs:line 110
   at Grynwald.MdDocs.ApiReference.Templates.Default.PageFactory..ctor(IApiReferencePathProvider pathProvider, ApiReferenceConfiguration configuration, AssemblyDocumentation assemblyDocumentation, ILogger logger) in /_/src/MdDocs.ApiReference/Templates/Default/PageFactory.cs:line 34
   at Grynwald.MdDocs.ApiReference.Templates.Default.PageFactory..ctor(IApiReferencePathProvider pathProvider, ApiReferenceConfiguration configuration, AssemblyDocumentation assemblyDocumentation) in /_/src/MdDocs.ApiReference/Templates/Default/PageFactory.cs:line 24
   at Grynwald.MdDocs.ApiReference.Templates.Default.ApiReferenceDefaultTemplate.Render(AssemblyDocumentation model) in /_/src/MdDocs.ApiReference/Templates/Default/ApiReferenceDefaultTemplate.cs:line 26
   at Grynwald.MdDocs.ApiReference.Commands.ApiReferenceCommand.Execute() in /_/src/MdDocs.ApiReference/Commands/ApiReferenceCommand.cs:line 32
   at Grynwald.MdDocs.Program.OnApiReferenceCommand(ILogger logger, ApiReferenceOptions opts) in /_/src/MdDocs/Program.cs:line 59
   at Grynwald.MdDocs.Program.<>c.<Main>b__0_1(ApiReferenceOptions opts) in /_/src/MdDocs/Program.cs:line 31
   at CommandLine.ParserResultExtensions.MapResult[T1,T2,TResult](ParserResult`1 result, Func`2 parsedFunc1, Func`2 parsedFunc2, Func`2 notParsedFunc)
   at Grynwald.MdDocs.Program.Main(String[] args) in /_/src/MdDocs/Program.cs:line 28
```

**Comment by ap0llo** [comment]
> 
> When I try to uninstall it I get an error saying it can't find the manifest file. 
> 

If `dotnet tool` is complaining about a missing manifest, it's trying to uninstall a local tool. You're probably missing the `--global` switch.

As for the error you're seeing, that's a new one I haven't seen before.
Would you mind opening a new issue and provide repro-steps or sample code for that error?

**Comment by ap0llo** [comment]
This should be fixed in the latest release 0.4.178  
https://github.com/ap0llo/mddocs/releases/tag/v0.4.178
